### PR TITLE
Fix/124

### PR DESCRIPTION
**Pull request by artalar**

### CODE REVIEW

File: packages/react/README.md [hunk 1]
```diff
@@ -13,40 +13,54 @@ Package for bindings [Reatom](https://github.com/artalar/reatom) store with Reac
...
+
 ```ts
-useAtom(atom: Atom, silent: boolean = true): AtomState
+const atomValue = useAtom(atom)
+```
+
+#### Depended value by selector
+
+```ts
+const atomValue = useAtom(atom, atomState => atomState[props.id], [props.id])
```
Comment (Wroud) [inline packages/react/README.md]: Can props been changed before callback was called?

File: packages/react/src/index.ts [hunk 2]
```diff
@@ -19,64 +20,91 @@ function useForceUpdate() {
...
+export function useAtom<T>(atom: Atom<T>): T
+export function useAtom<TI, TO = TI>(
+  atom: Atom<TI>,
+  mapper: (atomValue: TI) => TO,
+  deps: any[],
+): TO
+export function useAtom<TI, TO = TI>(
+  atom: Atom<TI>,
+  mapper: (atomValue: TI) => TO = (atomValue: TI) =>
+    (atomValue as unknown) as TO,
```
Comment (Wroud) [inline packages/react/src/index.ts]: mapper and deps must be optional here

Comment (Wroud) [reply packages/react/src/index.ts]: ... its already optional, sorry 🙏

File: packages/react/src/__tests__/index.tsx [hunk 3]
```diff
@@ -1,23 +1,193 @@
...
+  )
 }
 
 describe('@reatom/react', () => {
-  test('should throw if provider is not set', () => {
-    expect(() => render(<Counter />)).toThrowError(
-      '[reatom] The provider is not defined',
-    )
+  describe('useAtom', () => {
+    test('throw Error if provider is not set', () => {
```
Comment (belozer) [inline packages/react/src/__tests__/index.tsx]: What's about useAction?

File: packages/react/src/index.ts [hunk 4]
```diff
@@ -19,64 +20,91 @@ function useForceUpdate() {
...
+  deps: any[] = [],
```
Comment (belozer) [inline packages/react/src/index.ts]: `deps?: any[] = [],`

File: packages/react/src/index.ts [hunk 5]
```diff
@@ -19,64 +20,91 @@ function useForceUpdate() {
...
-): T {
-  const atomRef = useRef<Atom<T>>()
-  const stateRef = useRef<T>()
-  const unsubscribeRef = useRef<Function>()
-  const isMountRef = useRef(false)
+export function useAtom<T>(atom: Atom<T>): T
+export function useAtom<TI, TO = TI>(
+  atom: Atom<TI>,
+  mapper: (atomValue: TI) => TO,
+  deps: any[],
```
Comment (belozer) [inline packages/react/src/index.ts]: `deps?: any[] = [],`

File: packages/react/src/index.ts (hunk 4 above)
Comment (artalar) [reply packages/react/src/index.ts]: An argument with default value is optional (by typings) as is

File: packages/react/src/index.ts (hunk 5 above)
Comment (artalar) [reply packages/react/src/index.ts]: An argument with default value is optional (by typings) as is

**Pull request merged**